client/web-sveltekit/src/lib/repo/fileIcons.ts

```diff
@@ -0,0 +1,620 @@
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.OCAML: {
+            return {
+                value: DviClass.OCAML,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.PHP: {
+            return {
+                value: DviClass.PHP,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.PERL:
+        case FileExtension.PERL_PM: {
+            return {
+                value: DviClass.PERL,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.PNG: {
+            return {
+                value: mdiFilePngBox,
+                kind: Kind.MDI,
+            }
+        }
+        case FileExtension.PYTHON: {
+            return {
+                value: DviClass.PYTHON,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.R:
+        case FileExtension.R_CAP: {
+            return {
+                value: DviClass.R,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.RUBY:
+        case FileExtension.GEMFILE: {
+            return {
+                value: DviClass.RUBY,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.SCALA: {
+            return {
+                value: DviClass.SCALA,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.SASS: {
+            return {
+                value: DviClass.SASS,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.SVELTE: {
+            return {
+                value: DviClass.SVELTE,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.SVG: {
+            return {
+                value: mdiSvg,
+                kind: Kind.MDI,
+            }
+        }
+        case FileExtension.SWIFT: {
+            return {
+                value: DviClass.SWIFT,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.TYPESCRIPT: {
+            return {
+                value: DviClass.TYPESCRIPT,
+                kind: Kind.DEVICON,
+            }
+        }
+        case FileExtension.TEXT: {
+            return {
+                value: mdiText,
+                kind: Kind.MDI,
+            }
+        }
+        case FileExtension.YAML:
+        case FileExtension.YML: {
+            return {
+                value: mdiCog,
+                kind: Kind.MDI,
+            }
+        }
+        case FileExtension.ZIG: {
+            return {
+                value: DviClass.ZIG,
+                kind: Kind.DEVICON,
+            }
+        }
+        default: {
+            return {
+                value: mdiFileCode,
+                kind: Kind.MDI,
+            }
+        }
+    }
+}
+
+// only relevant for MDI icons
+const getColor = (extension: string): string => {
```

Comment:
This should be typed as `FileExtension` to get type checker support.

```suggestion
const getColor = (extension: FileExtension): string => {
```